emscripten.py

```diff
@@ -156,7 +156,7 @@ def apply_static_code_hooks(forwarded_json, code):
   code = shared.do_replace(code, '<<< ATINITS >>>', str(forwarded_json['ATINITS']))
   if settings.HAS_MAIN:
     code = shared.do_replace(code, '<<< ATMAINS >>>', str(forwarded_json['ATMAINS']))
-  if settings.EXIT_RUNTIME:
+  if settings.EXIT_RUNTIME and (not settings.MINIMAL_RUNTIME or settings.HAS_MAIN):
```

Comment:
I don't follow this part of the change?

Reply:
This is need for `asan.test_fs_no_main_minimal_runtime`.   Prior to this change this test (and many others) were not run with EXIT_RUNTIME enabled.

Without this part of the change the above test fails like:


```
$ ./tests/runner asan.test_fs_no_main_minimal_runtime --skip-slow
Test suites:
['test_core']
Running test_core: (1 tests)
(checking sanity from test runner)
shared:INFO: (Emscripten: Running sanity checks)
test_fs_no_main_minimal_runtime (test_core.asan) ... emcc: error: expected to find pattern in input JS: <<< ATEXITS >>>
```